strawberryfields/backends/tfbackend/backend.py

```diff
@@ -142,20 +142,20 @@ def prepare_vacuum_state(self, mode):
             remapped_mode = self._remap_modes(mode)
             self.circuit.prepare_vacuum_state(remapped_mode)
 
-    def prepare_coherent_state(self, alpha, mode):
+    def prepare_coherent_state(self, r, phi, mode):
         with tf.name_scope("Prepare_coherent"):
```

Comment:
is there a reference to understand what this `with` statements do?

Reply:
https://www.tensorflow.org/api_docs/python/tf/name_scope

It's to add the string "Prepare_coherent" to the names of the tensors created in that context. It allows for better tracebacks, it shows up on tensorboard, etc...